packages/frontend/src/components/staking/components/Validator.js

```diff
@@ -61,47 +73,54 @@ export default function Validator({
     currentValidators,
 }) {
     const [confirm, setConfirm] = useState(null);
-    const [farmList, setFarmList] = useState([]);
-    const [isFarmListLoading, setIsFarmListLoading] = useState(false);
     const nearAsFT = useNEARAsTokenWithMetadata();
     const accountId = useSelector(selectAccountId);
+
     const contractMetadataByContractId = useSelector(selectAllContractMetadata);
+    const tokenFiatValues = useSelector(selectTokensFiatValueUSD);
+    const tokenWhitelist = useSelector(selectTokenWhiteList);
 
     const dispatch = useDispatch();
     const stakeNotAllowed = !!selectedValidator && selectedValidator !== match.params.validator && !!currentValidators.length;
     const showConfirmModal = confirm === 'withdraw';
-    const stakingPoolHasFarms = validator && validator.version === ValidatorVersion[PROJECT_VALIDATOR_VERSION];
 
+    const [showClaimConfirmModal, setShowClaimConfirmModal] = useState(false);
+    const [selectedFarm, setSelectedFarm] = useState(null);
 
-    useEffect(() => {
-        const getFarms = async () => {
-            setIsFarmListLoading(true);
-
-            try {
-                const farms = await validator.contract.get_farms({ from_index: 0, limit: 300 });
-
-                const list = await Promise.all(farms.map(({ token_id }, i) => {
-                    dispatch(fetchToken({ contractName: token_id }));
-                    return validator.contract
-                        .get_unclaimed_reward({ account_id: accountId, farm_id: i })
-                        .catch(() => "0")
-                        .then((balance) => ({ token_id, balance, farm_id: i }));
-                }));
-
-                setFarmList(list);
-            } finally {
-                setIsFarmListLoading(false);
-            }
-        };
-        if (stakingPoolHasFarms) { getFarms(); }
-    }, [validator, stakingPoolHasFarms, accountId]);
+    const openModal = (farm) => {
+        setSelectedFarm(farm);
+        setShowClaimConfirmModal(true);
+    };
 
     const handleStakeAction = async () => {
         if (showConfirmModal && !loading) {
             await onWithdraw('withdraw', selectedValidator || validator.accountId);
             setConfirm('done');
         }
     };
+    const isFarmingValidator = validator?.version === FARMING_VALIDATOR_VERSION;
```

Comment:
Do we suppose that validator?.version is not optional param?

Reply:
There are a few cases when during the bootstrap of the component `validator` prop is still `undefined`. This way we can omit some errors and extra checks.